drivers/vfs/driver.go

```diff
@@ -241,6 +245,9 @@ func (d *Driver) dir(id string) string {
 
 // Remove deletes the content from the directory for a given id.
 func (d *Driver) Remove(id string) error {
+	if d.tempDirectory != nil {
+		return d.tempDirectory.Add(d.dir(id), time.Now().Format("20060102-150405"))
```

Comment:
`d.dir` might be in `d.imageStore` or `d.additionalHomes`, potentially on different filesystems; a `Rename` is not guaranteed to work.

(Applies also in other places, and it’s almost certainly out of scope of the prototype.)